public/js/length_visualisation.js

```diff
@@ -0,0 +1,339 @@
+        .data(function (d) { return d.data; })
+        .enter().append('rect')
+        .attr('class','bar')
+        .attr('data-toggle','tooltip')
+        .attr('title', function(i) {
+          return i.id+' '+self.prettify_evalue(i.evalue)+'<br>Length: '+i.value;
+        })
+        .attr('x', 1)
+        .attr('y', function(i) { return (self._y_scale(i.y0)); })
+        .attr('width', self._x_scale(this._bins[1].x) - self._x_scale(this._bins[0].x) - 1)
+        .attr('height', function (i) { return self._y_scale(i.y1) - self._y_scale(i.y0); })
+        .attr('fill', function(i) {
+          return i.color;
+        });
+  }
+
+  draw_query_line() {
+    var self = this;
+    var query_line = this._svg.d3.append('g')
+        .attr('class','query_line')
+        .attr('transform','translate('+(this._margin.left+this._x_scale(this._query.length))+',0)');
+
+    query_line.append('rect')
+        .attr('x',1)
+        .attr('width',4)
+        .attr('height',self._height)
+        .style('fill','rgb(55,0,232)');
+
+    query_line.append('text')
+        .attr('dy', '0.75em')
+        .attr('y', -5)
+        .attr('x', 5)
+        .attr('text-anchor','start')
+        .text('Query')
+        .style('fill','#000')
+        .attr('transform','rotate(-45)');
+  }
+
+  create_axis() {
+    var formatter = this.tick_formatter(this._seq_type.subject_seq_type);
+    // var formatter = this.visualisation_helpers.tick_formatter(this._x_scale,this._seq_type.subject_seq_type);
+    var x_axis = d3.svg.axis()
+        .scale(this._x_scale)
+        .orient('bottom')
+        .ticks(50)
+        .tickFormat(formatter);
+
+    var y_axis = d3.svg.axis()
+        .scale(this._y_scale)
+        .orient('left')
+        .tickFormat(function (e) {
+          if (Math.floor(e) != e) {
+            return ;
+          }
+          return e;
+        });
+
+    return {x: x_axis, y: y_axis};
+  }
+
+  draw_scales() {
+    var self = this;
+    var xContainer = this._svg.d3.append('g')
+        .attr('class', 'axis axis--x')
+        .attr('transform', 'translate('+self._margin.left+','+self._height+')')
+        .call(this._axis.x);
+
+    xContainer.selectAll('text').style('text-anchor','end')
+        .attr('x', '-8px')
+        .attr('y', '3px')
+        .attr('dy', '0')
+        .attr('transform','rotate(-90)');
+
+    var yContainer = this._svg.d3.append('g')
+        .attr('class','axis axis--y')
+        .attr('transform','translate('+self._margin.left+',0)')
+        .call(this._axis.y);
+  }
+
+  create_labels() {
+    var self = this;
+    this._svg.d3.append('text')
+        .attr('class','xaxis-label')
+        .attr('x',0.35 * self._width)
+        .attr('y',self._height + 65)
+        .text('Sequence Length');
+
+    this._svg.d3.append('text')
+        .attr('class', 'yaxis-label')
+        .attr('x',-255)
+        .attr('y',-15)
+        .attr('transform','rotate(-90)')
+        .text('Number of Sequences');
+  }
+
+  draw_legend() {
```

Comment:
Is this the exact same code from graphical overview? If so it's safe to remove it I guess since we are not drawing a legend here.